### PR TITLE
Feature/add sdk identifier

### PR DESCRIPTION
**Pull request by denis-zyk**
@UmanShahzad Please review :-)

### CODE REVIEW

File: internal/utils.go [hunk 1]
```diff
@@ -3,9 +3,37 @@ package internal
 import (
 	"fmt"
 	"net/url"
+	"runtime"
+	"runtime/debug"
 	"strings"
 )
 
+var (
+	// SdkVersion could be set via build flags `-ldflags="-X internal.SdkVersion=${CI_COMMIT_TAG:-$CI_COMMIT_SHA}"`
```
Comment (UmanShahzad) [inline internal/utils.go:12]: This isn't good Go practice, as no one's going to want to override it this way. If there really needs to be open support for changing the user-agent, then it should be via a publicly available function like the currently-internal `WithSdkIdentifier` that helps to build up the request object with custom parameters, one of which could be the user-agent.

**Review by UmanShahzad — COMMENTED** [review]
All good; if the user-agent doesn't need to be actually changeable easily then this is fine. My own experience is users don't often want to change it, so should be OK, but in the future a request may come in .

**Comment by oxy-nr** [comment]
LGTM

**Review by oxy-gerach — APPROVED** [review]
LGTM

**Review by oxy-nr — APPROVED** [review]

**Pull request merged**